evap/staff/importers.py

```diff
@@ -476,6 +496,47 @@ def check_evaluation_data_correctness(self, semester):
                 ).format(course_type_name)
             )
 
+    def set_existing_course_to_merge_or_add_errors(self, semester, evaluation_data):
+        merge_candidate = Course.objects.filter(
+            semester=semester, name_de=evaluation_data.name_de, name_en=evaluation_data.name_en
+        ).first()
+        if merge_candidate:
+            merge_hindrances = evaluation_data.get_course_merge_hindrances(merge_candidate)
+            if merge_hindrances:
+                self.errors[ImporterError.COURSE].append(
+                    format_html(
+                        _(
+                            "Course {} ({}) does already exist in this semester, but the courses can not be merged for the following reasons:{}."
+                        ).format(
+                            evaluation_data.name_en,
+                            evaluation_data.name_de,
+                            "".join(f"<br /> - {msg}" for msg in merge_hindrances),
+                        )
+                    )
+                )
+            else:
+                evaluation_data.existing_course = merge_candidate
+                self.warnings[ImporterWarning.EXISTS].append(
+                    _(
+                        "Course {} ({}) already exists with identical attributes. Course is not created and users are put into the evaluation of that course."
```

Comment:
Two small things:
- "identical attributes" might be misleading, as degrees could differ.
- The message does not say that degrees will be merged.

@janno42 is this ok, or do we want a different message?

Reply:
Both things should optimally be included in the warning message :)